src/distributions/dragonwell/installer.ts

```diff
@@ -0,0 +1,173 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+
+import fs from 'fs';
+import path from 'path';
+
+import {JavaBase} from '../base-installer';
+import {extractJdkFile, getDownloadArchiveExtension} from '../../util';
+import {IDragonwellVersions, IDragonwellAllVersions} from './models';
+import {
+  JavaDownloadRelease,
+  JavaInstallerOptions,
+  JavaInstallerResults
+} from '../base-models';
+
+export class DragonwellDistribution extends JavaBase {
+  constructor(installerOptions: JavaInstallerOptions) {
+    super('Dragonwell', installerOptions);
+  }
+
+  protected async findPackageForDownload(
+    version: string
+  ): Promise<JavaDownloadRelease> {
+    if (!this.stable) {
+      throw new Error('Early access versions are not supported');
+    }
+    let majorVersion = version;
+    if (version.includes('.')) {
+      const splits = version.split('.');
+      majorVersion = splits[0];
+      version = splits.length >= 3 ? splits.slice(0, 3).join('.') : version;
+    }
+    const edition = majorVersion == '17' ? 'Standard' : 'Extended';
+    const availableVersions = await this.getAvailableVersions();
+    const matchedVersions = availableVersions
+      .filter(item => item.jdk_version == version && item.edition == edition)
+      .map(item => {
+        return {
+          version: item.jdk_version,
+          url: item.download_link
+        } as JavaDownloadRelease;
+      });
+    const resolvedVersion =
+      matchedVersions.length > 0 ? matchedVersions[0] : null;
+    if (!resolvedVersion) {
+      const versionsMsg = core.isDebug()
+        ? `Available versions: ${availableVersions}`
+        : '';
+      throw new Error(
+        `Cannot find satisfied version for ${version}.${versionsMsg}`
+      );
+    }
+    return resolvedVersion;
```

Comment:
```suggestion
    if (!matchedVersions.length) {
        throw new Error(
        `Couldn't find any satisfied version for the specified: "${version}".`
      );
    }
    const resolvedVersion = matchedVersions[0];
    return resolvedVersion;
```
I'd suggest to simplify the code. You can incorporate the list of available versions to the error message if you wish.

Reply:
> I'd suggest to simplify the code. You can incorporate the list of available versions to the error message if you wish.

I've update these lines of code as your suggested.